embark-consult.el

```diff
@@ -103,44 +111,89 @@
-      (dolist (line lines)
-        (pcase-let*
-            ((`(,loc . ,num) (consult--get-location line))
-             ;; the text properties added to the following strings are
-             ;; taken from occur-engine
-             (lineno (propertize (format "%7d:" num)
-                                 'occur-prefix t
-                                 ;; Allow insertion of text at the end
-                                 ;; of the prefix (for Occur Edit mode).
-                                 'front-sticky t
-                                 'rear-nonsticky t
+    (dolist (line lines)
+      (pcase-let*
+          ((`(,loc . ,num) (consult--get-location line))
+           ;; the text properties added to the following strings are
+           ;; taken from occur-engine
+           (lineno (propertize (format "%7d:" num)
+                               'occur-prefix t
+                               ;; Allow insertion of text at the end
+                               ;; of the prefix (for Occur Edit mode).
+                               'front-sticky t
+                               'rear-nonsticky t
+                               'occur-target loc
+                               'follow-link t
+                               'help-echo mouse-msg))
+           (contents (propertize (embark-consult--strip line)
                                  'occur-target loc
+                                 'occur-match t
                                  'follow-link t
                                  'help-echo mouse-msg))
-             (contents (propertize (embark-consult--strip line)
-                                   'occur-target loc
-                                   'occur-match t
-                                   'follow-link t
-                                   'help-echo mouse-msg))
-             (nl (propertize "\n" 'occur-target loc))
-             (this-buf (marker-buffer loc)))
+           (nl (propertize "\n" 'occur-target loc))
+           (this-buf (marker-buffer loc)))
+        (unless (eq this-buf last-buf)
+          (insert (propertize
+                   (format "lines from buffer: %s\n" this-buf)
+                   'face list-matching-lines-buffer-name-face))
+          (setq last-buf this-buf))
+        (insert (concat lineno contents nl))))))
+
+(defun embark-consult--location-to-grep-results-writer (lines)
+  "Format the results from `consult-location' range of commands for the `grep-mode' as alternative to occur.
+You can edit results with `wgrep' in grep buffer including line deletion and insertion of new lines.
+The elements of LINES are assumed to be values of category `consult-location'."
+  (let (last-buf
+        filename)
+    (insert (propertize "Caution:" 'wgrep-header t 'font-lock-face '(:inherit 'compilation-warning :underline t))
+            (propertize " only buffers backed by a file are exported to grep-mode.\n" 'wgrep-header t 'font-lock-face '(:inherit 'compilation-warning)))
```

Comment:
The caution note isn't a bad idea, but I would prefer something less verbose, 
instead of "Exported Consult results\n\n" what about "Exported line results (file-backed buffers only):\n\n"?